File: devnhwa/2024_02_27/Lv.0/문자열의 뒤의 n글자.java

/*
https://school.programmers.co.kr/learn/courses/30/lessons/181910

문제 설명
문자열 my_string과 정수 n이 매개변수로 주어질 때, my_string의 뒤의 n글자로 이루어진 문자열을 return 하는 solution 함수를 작성해 주세요.

제한사항
my_string은 숫자와 알파벳으로 이루어져 있습니다.
1 ≤ my_string의 길이 ≤ 1,000
1 ≤ n ≤ my_string의 길이
입출력 예
my_string	n	result
"ProgrammerS123"	11	"grammerS123"
"He110W0r1d"	5	"W0r1d"
입출력 예
입출력 예 #1

예제 1번의 my_string에서 뒤의 11글자는 "grammerS123"이므로 이 문자열을 return 합니다.
입출력 예 #2

예제 2번의 my_string에서 뒤의 5글자는 "W0r1d"이므로 이 문자열을 return 합니다.

*/
class Solution {
    public String solution(String my_string, int n) {
        String answer = "";
        answer = my_string.substring(my_string.length()-n, my_string.length());
        return answer;
    }
}

File: decembaek/2024_03/2024_03_05/문자열의_뒤의_n글자.py
"""
https://school.programmers.co.kr/learn/courses/30/lessons/181910

프로그래머스 Lv.0

제목 : 문자열의 뒤의 n 글자
"""


# 풀이 시작
def solution(my_string, n):
    answer = ""
    return answer


# decembaek 풀이 (만족)
def solution(my_string, n):
    return my_string[-n:]


# 다른사람 풀이
def solution(my_string, n):
    return my_string[-n:]


solution = lambda my_string, n: my_string[len(my_string) - n :]
